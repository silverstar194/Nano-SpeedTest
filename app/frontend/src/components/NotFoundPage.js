import React from 'react';
import Header from './Header';

export default() => (
    <div>
        <Header/>
        <h1 className="page-header text-center">404</h1>
    </div>
);