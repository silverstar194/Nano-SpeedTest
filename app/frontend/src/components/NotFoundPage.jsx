import React from 'react';

export default() => (
 <div class="container">
    <div class="row">
        <div class="col-md-12 text-center">
            <div class="error-template">
                <h1>
                    Oops!</h1>
                <h2>
                    404 Not Found</h2>
                <div class="error-details">
                    Sorry, an error has occured, Requested page not found!
                </div>
                <div class="error-actions">
                    <a href="/" class="btn btn-primary btn-lg"><span class="glyphicon glyphicon-home"></span>
                        Take Me Home</a>
                </div>
            </div>
        </div>
    </div>
</div>
);