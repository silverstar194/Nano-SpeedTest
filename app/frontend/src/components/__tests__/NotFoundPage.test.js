import React from 'react';
import renderer from 'react-test-renderer';
import NotFoundPage from '../NotFoundPage';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter} from 'react-router-dom';
import { shallow } from 'enzyme';

const mockStore = configureStore();
let store = mockStore({
    navigation: {
        activeTab: ''
    }
});

const props = {
	history: {}
};

it('shallow renders without crashing', () => {
    shallow(<NotFoundPage store={store} {...props}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <NotFoundPage {...props}/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
