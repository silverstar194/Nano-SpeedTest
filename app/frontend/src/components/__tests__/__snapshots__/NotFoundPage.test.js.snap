// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
         
        Speed Test
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Stats"
        onClick={[Function]}
      >
         
        Statistics
         
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link "
        href="/Info"
        onClick={[Function]}
      >
         
        More Info
         
      </a>
    </li>
  </div>
  <h1
    className="page-header text-center"
  >
    404
  </h1>
</div>
`;
